src/main/lens-proxy.ts

```diff
@@ -73,12 +71,41 @@ export class LensProxy {
     if (cluster) {
       const proxyUrl = await cluster.contextHandler.resolveAuthProxyUrl() + req.url.replace(apiKubePrefix, "")
       const apiUrl = url.parse(cluster.apiUrl)
-      const res = new http.ServerResponse(req)
-      res.assignSocket(socket)
-      res.setHeader("Location", proxyUrl)
-      res.setHeader("Host", apiUrl.hostname)
-      res.statusCode = 302
-      res.end()
+      const pUrl = url.parse(proxyUrl)
+      const connectOpts = {port: parseInt(pUrl.port), host: pUrl.hostname}
+      const proxySocket = new net.Socket()
+      proxySocket.connect(connectOpts, () => {
+        proxySocket.write(`${req.method} ${pUrl.path} HTTP/1.1\r\n`)
+        proxySocket.write(`Host: ${apiUrl.host}\r\n`)
+        for(let i = 0; i < req.rawHeaders.length; i++) {
+          const key = req.rawHeaders[i]
+          if (key !== "Host" && key !== "Authorization") {
+            proxySocket.write(`${req.rawHeaders[i]}: ${req.rawHeaders[i+1]}\r\n`)
+          }
+          i++
+        }
+        proxySocket.write("\r\n")
+        proxySocket.write(head)
+      })
+      proxySocket.on('data', function (chunk) {
+        socket.write(chunk)
+      })
+      proxySocket.on('end', function () {
+        socket.end()
+      })
+      proxySocket.on('error', function (err) {
+        socket.write("HTTP/" + req.httpVersion + " 500 Connection error\r\n\r\n");
```

Comment:
Should we log the error some where, maybe to console?

Reply:
Probably not here, proxy itself will log (and retry) if response has an error.